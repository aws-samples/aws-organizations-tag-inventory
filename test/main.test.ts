/*
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy of this
 * software and associated documentation files (the "Software"), to deal in the Software
 * without restriction, including without limitation the rights to use, copy, modify,
 * merge, publish, distribute, sublicense, and/or sell copies of the Software, and to
 * permit persons to whom the Software is furnished to do so.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
 * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
 * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
 * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
 * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
 * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

import { App } from "aws-cdk-lib";
import { Template } from "aws-cdk-lib/assertions";
// @ts-ignore
import { SpokeStack } from "../src/stacks/SpokeStack";

test("Snapshot", () => {
  const app = new App();
  const stack = new SpokeStack(app, "test", {
    enabledRegions: "us-east-1,us-east-2,us-west-1,us-west-2",
    aggregatorRegion: "us-east-2",
    bucketName: "test-bucket",
    centralRoleArn: "test-role",
  });

  const template = Template.fromStack(stack);
  expect(template.toJSON()).toMatchSnapshot();
});
