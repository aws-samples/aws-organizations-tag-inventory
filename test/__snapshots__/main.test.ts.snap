// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot 1`] = `
{
  "Conditions": {
    "spokeScheduleDailyCondition2496F6E6": {
      "Fn::Equals": [
        {
          "Ref": "ScheduleParameter",
        },
        "DAILY",
      ],
    },
    "spokeScheduleMonthlyConditionCA0116AA": {
      "Fn::Equals": [
        {
          "Ref": "ScheduleParameter",
        },
        "MONTHLY",
      ],
    },
    "spokeScheduleWeeklyCondition205254D0": {
      "Fn::Equals": [
        {
          "Ref": "ScheduleParameter",
        },
        "WEEKLY",
      ],
    },
  },
  "Mappings": {
    "ServiceprincipalMap": {
      "af-south-1": {
        "states": "states.af-south-1.amazonaws.com",
      },
      "ap-east-1": {
        "states": "states.ap-east-1.amazonaws.com",
      },
      "ap-northeast-1": {
        "states": "states.ap-northeast-1.amazonaws.com",
      },
      "ap-northeast-2": {
        "states": "states.ap-northeast-2.amazonaws.com",
      },
      "ap-northeast-3": {
        "states": "states.ap-northeast-3.amazonaws.com",
      },
      "ap-south-1": {
        "states": "states.ap-south-1.amazonaws.com",
      },
      "ap-south-2": {
        "states": "states.ap-south-2.amazonaws.com",
      },
      "ap-southeast-1": {
        "states": "states.ap-southeast-1.amazonaws.com",
      },
      "ap-southeast-2": {
        "states": "states.ap-southeast-2.amazonaws.com",
      },
      "ap-southeast-3": {
        "states": "states.ap-southeast-3.amazonaws.com",
      },
      "ap-southeast-4": {
        "states": "states.ap-southeast-4.amazonaws.com",
      },
      "ca-central-1": {
        "states": "states.ca-central-1.amazonaws.com",
      },
      "cn-north-1": {
        "states": "states.cn-north-1.amazonaws.com",
      },
      "cn-northwest-1": {
        "states": "states.cn-northwest-1.amazonaws.com",
      },
      "eu-central-1": {
        "states": "states.eu-central-1.amazonaws.com",
      },
      "eu-central-2": {
        "states": "states.eu-central-2.amazonaws.com",
      },
      "eu-north-1": {
        "states": "states.eu-north-1.amazonaws.com",
      },
      "eu-south-1": {
        "states": "states.eu-south-1.amazonaws.com",
      },
      "eu-south-2": {
        "states": "states.eu-south-2.amazonaws.com",
      },
      "eu-west-1": {
        "states": "states.eu-west-1.amazonaws.com",
      },
      "eu-west-2": {
        "states": "states.eu-west-2.amazonaws.com",
      },
      "eu-west-3": {
        "states": "states.eu-west-3.amazonaws.com",
      },
      "il-central-1": {
        "states": "states.il-central-1.amazonaws.com",
      },
      "me-central-1": {
        "states": "states.me-central-1.amazonaws.com",
      },
      "me-south-1": {
        "states": "states.me-south-1.amazonaws.com",
      },
      "sa-east-1": {
        "states": "states.sa-east-1.amazonaws.com",
      },
      "us-east-1": {
        "states": "states.us-east-1.amazonaws.com",
      },
      "us-east-2": {
        "states": "states.us-east-2.amazonaws.com",
      },
      "us-gov-east-1": {
        "states": "states.us-gov-east-1.amazonaws.com",
      },
      "us-gov-west-1": {
        "states": "states.us-gov-west-1.amazonaws.com",
      },
      "us-iso-east-1": {
        "states": "states.amazonaws.com",
      },
      "us-iso-west-1": {
        "states": "states.amazonaws.com",
      },
      "us-isob-east-1": {
        "states": "states.amazonaws.com",
      },
      "us-west-1": {
        "states": "states.us-west-1.amazonaws.com",
      },
      "us-west-2": {
        "states": "states.us-west-2.amazonaws.com",
      },
    },
  },
  "Metadata": {
    "cdk_nag": {
      "rules_to_suppress": [
        {
          "id": "AwsSolutions-IAM4",
          "reason": "AWS managed policies acceptable for sample",
        },
        {
          "id": "AwsSolutions-IAM5",
          "reason": "Wildcard permissions have been scoped down",
        },
        {
          "id": "AwsSolutions-L1",
          "reason": "Manually managing versions",
        },
      ],
    },
  },
  "Outputs": {
    "spokeMyIndextaginventoryallresourcesarnoutput76D45A96": {
      "Description": "Arn of tag-inventory-all-resources view",
      "Value": {
        "Fn::GetAtt": [
          "spokeMyIndexresourceexplorerindexcustomresource40170C84",
          "ViewArn",
        ],
      },
    },
  },
  "Parameters": {
    "AggregatorRegionParameter": {
      "AllowedValues": [
        "af-south-1",
        "ap-east-1",
        "ap-northeast-1",
        "ap-northeast-2",
        "ap-northeast-3",
        "ap-south-1",
        "ap-south-2",
        "ap-southeast-1",
        "ap-southeast-2",
        "ap-southeast-3",
        "ap-southeast-4",
        "ca-central-1",
        "cn-north-1",
        "cn-northwest-1",
        "eu-central-1",
        "eu-central-2",
        "eu-north-1",
        "eu-south-1",
        "eu-south-2",
        "eu-west-1",
        "eu-west-2",
        "eu-west-3",
        "il-central-1",
        "me-central-1",
        "me-south-1",
        "sa-east-1",
        "us-east-1",
        "us-east-2",
        "us-gov-east-1",
        "us-gov-west-1",
        "us-iso-east-1",
        "us-iso-west-1",
        "us-isob-east-1",
        "us-west-1",
        "us-west-2",
      ],
      "Default": "us-east-2",
      "Description": "The region that contains teh Resource Explorer aggregator",
      "Type": "String",
    },
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "BucketNameParameter": {
      "Default": "test-bucket",
      "Description": "Name of the central account bucket where tag inventory data is stored",
      "Type": "String",
    },
    "CentralRoleArnParameter": {
      "Default": "test-role",
      "Description": "ARN of the central account's cross account role with permissions to write to the centralized bucket where tag inventory data is stored",
      "Type": "String",
    },
    "EnabledRegionsParameter": {
      "Default": "us-east-1,us-east-2,us-west-1,us-west-2",
      "Description": "Regions to enable Resource Explorer Indexing",
      "Type": "CommaDelimitedList",
    },
    "OrganizationPayerAccountIdParameter": {
      "Description": "The id of the AWS organization payer account",
      "Type": "String",
    },
    "ScheduleParameter": {
      "AllowedValues": [
        "DAILY",
        "WEEKLY",
        "MONTHLY",
      ],
      "Description": "The frequency jobs are run",
      "Type": "String",
    },
    "TopicArnParameter": {
      "Description": "ARN of the central account's notification topic",
      "Type": "String",
    },
  },
  "Resources": {
    "comamazonawscdkcustomresourcesresourceexplorerindexproviderframeworkonEventCA9D2DDB": {
      "DependsOn": [
        "comamazonawscdkcustomresourcesresourceexplorerindexproviderframeworkonEventServiceRoleDefaultPolicy196222C7",
        "comamazonawscdkcustomresourcesresourceexplorerindexproviderframeworkonEventServiceRole6629EB69",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "7382a0addb9f34974a1ea6c6c9b063882af874828f366f5c93b2b7b64db15c94.zip",
        },
        "Description": "AWS CDK resource provider framework - onEvent (test/com.amazonaws.cdk.custom-resources.resource-explorer-index-provider/resource-explorer-index-provider)",
        "Environment": {
          "Variables": {
            "USER_ON_EVENT_FUNCTION_ARN": {
              "Fn::GetAtt": [
                "comamazonawscdkcustomresourcesresourceexplorerindexproviderresourceexplorerindexfnE4EA2129",
                "Arn",
              ],
            },
          },
        },
        "Handler": "framework.onEvent",
        "Role": {
          "Fn::GetAtt": [
            "comamazonawscdkcustomresourcesresourceexplorerindexproviderframeworkonEventServiceRole6629EB69",
            "Arn",
          ],
        },
        "Runtime": "nodejs18.x",
        "Tags": [
          {
            "Key": "Solution",
            "Value": "aws-organizations-tag-inventory",
          },
          {
            "Key": "Url",
            "Value": "https://github.com/aws-samples/aws-organizations-tag-inventory",
          },
        ],
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "comamazonawscdkcustomresourcesresourceexplorerindexproviderframeworkonEventServiceRole6629EB69": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
        "Tags": [
          {
            "Key": "Solution",
            "Value": "aws-organizations-tag-inventory",
          },
          {
            "Key": "Url",
            "Value": "https://github.com/aws-samples/aws-organizations-tag-inventory",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "comamazonawscdkcustomresourcesresourceexplorerindexproviderframeworkonEventServiceRoleDefaultPolicy196222C7": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "lambda:InvokeFunction",
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "comamazonawscdkcustomresourcesresourceexplorerindexproviderresourceexplorerindexfnE4EA2129",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "comamazonawscdkcustomresourcesresourceexplorerindexproviderresourceexplorerindexfnE4EA2129",
                          "Arn",
                        ],
                      },
                      ":*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "comamazonawscdkcustomresourcesresourceexplorerindexproviderframeworkonEventServiceRoleDefaultPolicy196222C7",
        "Roles": [
          {
            "Ref": "comamazonawscdkcustomresourcesresourceexplorerindexproviderframeworkonEventServiceRole6629EB69",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "comamazonawscdkcustomresourcesresourceexplorerindexproviderresourceexplorerindexfnE4EA2129": {
      "DependsOn": [
        "comamazonawscdkcustomresourcesresourceexplorerindexproviderresourceexplorerindexfnServiceRoleDefaultPolicy74BEE050",
        "comamazonawscdkcustomresourcesresourceexplorerindexproviderresourceexplorerindexfnServiceRole2F22B68A",
      ],
      "Properties": {
        "Architectures": [
          "arm64",
        ],
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "f41ea6430644aad7f00b09d3046107637386f03257192b1e1dc88eee1500e7dd.zip",
        },
        "Environment": {
          "Variables": {
            "AWS_NODEJS_CONNECTION_REUSE_ENABLED": "1",
          },
        },
        "Handler": "index.onEvent",
        "Layers": [
          {
            "Ref": "spokelayerslayerBCD68A79",
          },
        ],
        "Role": {
          "Fn::GetAtt": [
            "comamazonawscdkcustomresourcesresourceexplorerindexproviderresourceexplorerindexfnServiceRole2F22B68A",
            "Arn",
          ],
        },
        "Runtime": "nodejs20.x",
        "Tags": [
          {
            "Key": "Solution",
            "Value": "aws-organizations-tag-inventory",
          },
          {
            "Key": "Url",
            "Value": "https://github.com/aws-samples/aws-organizations-tag-inventory",
          },
        ],
        "Timeout": 60,
      },
      "Type": "AWS::Lambda::Function",
    },
    "comamazonawscdkcustomresourcesresourceexplorerindexproviderresourceexplorerindexfnServiceRole2F22B68A": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
        "Tags": [
          {
            "Key": "Solution",
            "Value": "aws-organizations-tag-inventory",
          },
          {
            "Key": "Url",
            "Value": "https://github.com/aws-samples/aws-organizations-tag-inventory",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "comamazonawscdkcustomresourcesresourceexplorerindexproviderresourceexplorerindexfnServiceRoleDefaultPolicy74BEE050": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "resource-explorer-2:CreateView",
                "resource-explorer-2:ListViews",
                "resource-explorer-2:GetView",
                "resource-explorer-2:GetIndex",
                "resource-explorer-2:CreateIndex",
                "resource-explorer-2:UpdateIndexType",
                "resource-explorer-2:GetDefaultView",
                "resource-explorer-2:AssociateDefaultView",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
            {
              "Action": "iam:CreateServiceLinkedRole",
              "Effect": "Allow",
              "Resource": "arn:aws:iam::*:role/aws-service-role/resource-explorer-2.amazonaws.com/AWSServiceRoleForResourceExplorer",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "comamazonawscdkcustomresourcesresourceexplorerindexproviderresourceexplorerindexfnServiceRoleDefaultPolicy74BEE050",
        "Roles": [
          {
            "Ref": "comamazonawscdkcustomresourcesresourceexplorerindexproviderresourceexplorerindexfnServiceRole2F22B68A",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "spokeMergeResultsfn262D356D": {
      "DependsOn": [
        "spokeMergeResultsfnServiceRole07998249",
      ],
      "Properties": {
        "Architectures": [
          "arm64",
        ],
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "b3fdc195f7b82de4226da4f91b9890e6766a77f12ec2b737a846ecb0ddeb640b.zip",
        },
        "Environment": {
          "Variables": {
            "AWS_NODEJS_CONNECTION_REUSE_ENABLED": "1",
            "LOG_LEVEL": "DEBUG",
          },
        },
        "Handler": "index.onEvent",
        "Layers": [
          {
            "Fn::Join": [
              "",
              [
                "arn:aws:lambda:",
                {
                  "Ref": "AWS::Region",
                },
                ":094274105915:layer:AWSLambdaPowertoolsTypeScript:11",
              ],
            ],
          },
        ],
        "Role": {
          "Fn::GetAtt": [
            "spokeMergeResultsfnServiceRole07998249",
            "Arn",
          ],
        },
        "Runtime": "nodejs20.x",
        "Tags": [
          {
            "Key": "Solution",
            "Value": "aws-organizations-tag-inventory",
          },
          {
            "Key": "Url",
            "Value": "https://github.com/aws-samples/aws-organizations-tag-inventory",
          },
        ],
        "Timeout": 60,
      },
      "Type": "AWS::Lambda::Function",
    },
    "spokeMergeResultsfnServiceRole07998249": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
        "Tags": [
          {
            "Key": "Solution",
            "Value": "aws-organizations-tag-inventory",
          },
          {
            "Key": "Url",
            "Value": "https://github.com/aws-samples/aws-organizations-tag-inventory",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "spokeMyIndexresourceexplorerindexcustomresource40170C84": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "AGGREGATOR_INDEX_REGION": {
          "Ref": "AggregatorRegionParameter",
        },
        "ENABLED_REGIONS": {
          "Ref": "EnabledRegionsParameter",
        },
        "ServiceToken": {
          "Fn::GetAtt": [
            "comamazonawscdkcustomresourcesresourceexplorerindexproviderframeworkonEventCA9D2DDB",
            "Arn",
          ],
        },
      },
      "Type": "Custom::ResourceExplorerIndex",
      "UpdateReplacePolicy": "Retain",
    },
    "spokeSchedulerFAB0B53F": {
      "Properties": {
        "FlexibleTimeWindow": {
          "MaximumWindowInMinutes": 60,
          "Mode": "FLEXIBLE",
        },
        "Name": "TagInventorySchedule",
        "ScheduleExpression": {
          "Fn::If": [
            "spokeScheduleDailyCondition2496F6E6",
            "cron(0 1 ? * * *)",
            {
              "Fn::If": [
                "spokeScheduleWeeklyCondition205254D0",
                "cron(0 1 ? * SAT *)",
                {
                  "Fn::If": [
                    "spokeScheduleMonthlyConditionCA0116AA",
                    "cron(0 1 ? 1/1 SAT#4 *)",
                    "cron(0 1 ? * SAT *)",
                  ],
                },
              ],
            },
          ],
        },
        "ScheduleExpressionTimezone": "America/New_York",
        "State": "ENABLED",
        "Target": {
          "Arn": {
            "Ref": "spokeSpokeAccountStateMachineD18081B4",
          },
          "RoleArn": {
            "Fn::GetAtt": [
              "spokeSchedulerRole9550E8FD",
              "Arn",
            ],
          },
        },
      },
      "Type": "AWS::Scheduler::Schedule",
    },
    "spokeSchedulerRole9550E8FD": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "scheduler.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "Tags": [
          {
            "Key": "Solution",
            "Value": "aws-organizations-tag-inventory",
          },
          {
            "Key": "Url",
            "Value": "https://github.com/aws-samples/aws-organizations-tag-inventory",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "spokeSchedulerRoleDefaultPolicy3D206F1D": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "states:StartExecution",
              "Effect": "Allow",
              "Resource": {
                "Ref": "spokeSpokeAccountStateMachineD18081B4",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "spokeSchedulerRoleDefaultPolicy3D206F1D",
        "Roles": [
          {
            "Ref": "spokeSchedulerRole9550E8FD",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "spokeSearchfn7ED063A8": {
      "DependsOn": [
        "spokeSearchfnServiceRoleDefaultPolicyFFFA2302",
        "spokeSearchfnServiceRole7D05723F",
      ],
      "Properties": {
        "Architectures": [
          "arm64",
        ],
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "f7fd84e12658d277aefbf1bae3688f7aaef6c52c0708f0caae9370118e8416fe.zip",
        },
        "Environment": {
          "Variables": {
            "AWS_NODEJS_CONNECTION_REUSE_ENABLED": "1",
            "LOG_LEVEL": "DEBUG",
            "VIEW_ARN": {
              "Fn::GetAtt": [
                "spokeMyIndexresourceexplorerindexcustomresource40170C84",
                "ViewArn",
              ],
            },
          },
        },
        "Handler": "index.onEvent",
        "Layers": [
          {
            "Ref": "spokelayerslayerBCD68A79",
          },
          {
            "Fn::Join": [
              "",
              [
                "arn:aws:lambda:",
                {
                  "Ref": "AWS::Region",
                },
                ":094274105915:layer:AWSLambdaPowertoolsTypeScript:11",
              ],
            ],
          },
        ],
        "Role": {
          "Fn::GetAtt": [
            "spokeSearchfnServiceRole7D05723F",
            "Arn",
          ],
        },
        "Runtime": "nodejs20.x",
        "Tags": [
          {
            "Key": "Solution",
            "Value": "aws-organizations-tag-inventory",
          },
          {
            "Key": "Url",
            "Value": "https://github.com/aws-samples/aws-organizations-tag-inventory",
          },
        ],
        "Timeout": 60,
      },
      "Type": "AWS::Lambda::Function",
    },
    "spokeSearchfnServiceRole7D05723F": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
        "Tags": [
          {
            "Key": "Solution",
            "Value": "aws-organizations-tag-inventory",
          },
          {
            "Key": "Url",
            "Value": "https://github.com/aws-samples/aws-organizations-tag-inventory",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "spokeSearchfnServiceRoleDefaultPolicyFFFA2302": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "resource-explorer-2:Search",
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "spokeSearchfnServiceRoleDefaultPolicyFFFA2302",
        "Roles": [
          {
            "Ref": "spokeSearchfnServiceRole7D05723F",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "spokeSpokeAccountStateMachineD18081B4": {
      "DeletionPolicy": "Delete",
      "DependsOn": [
        "spokeSpokeAccountStateMachineRoleDefaultPolicy4B03EE37",
        "spokeSpokeAccountStateMachineRoleEF6757B7",
      ],
      "Properties": {
        "DefinitionS3Location": {
          "Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "Key": "50b49bfbcce0a69ef0a1eea4ec522504f517ec57bb2a630243f3eb7110c21312.json",
        },
        "DefinitionSubstitutions": {
          "CENTRAL_BUCKET_NAME": {
            "Ref": "BucketNameParameter",
          },
          "CENTRAL_ROLE_ARN": {
            "Ref": "CentralRoleArnParameter",
          },
          "MERGE_FUNCTION": {
            "Fn::GetAtt": [
              "spokeMergeResultsfn262D356D",
              "Arn",
            ],
          },
          "SEARCH_FUNCTION": {
            "Fn::GetAtt": [
              "spokeSearchfn7ED063A8",
              "Arn",
            ],
          },
          "TOPIC_ARN": {
            "Ref": "TopicArnParameter",
          },
        },
        "LoggingConfiguration": {
          "Destinations": [
            {
              "CloudWatchLogsLogGroup": {
                "LogGroupArn": {
                  "Fn::GetAtt": [
                    "spokeSpokeAccountStateMachineSpokeAccountStateMachineLogGroupBF1216DD",
                    "Arn",
                  ],
                },
              },
            },
          ],
          "IncludeExecutionData": true,
          "Level": "ALL",
        },
        "RoleArn": {
          "Fn::GetAtt": [
            "spokeSpokeAccountStateMachineRoleEF6757B7",
            "Arn",
          ],
        },
        "Tags": [
          {
            "Key": "Solution",
            "Value": "aws-organizations-tag-inventory",
          },
          {
            "Key": "Url",
            "Value": "https://github.com/aws-samples/aws-organizations-tag-inventory",
          },
        ],
        "TracingConfiguration": {
          "Enabled": true,
        },
      },
      "Type": "AWS::StepFunctions::StateMachine",
      "UpdateReplacePolicy": "Delete",
    },
    "spokeSpokeAccountStateMachineRoleDefaultPolicy4B03EE37": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "logs:CreateLogDelivery",
                "logs:GetLogDelivery",
                "logs:UpdateLogDelivery",
                "logs:DeleteLogDelivery",
                "logs:ListLogDeliveries",
                "logs:PutResourcePolicy",
                "logs:DescribeResourcePolicies",
                "logs:DescribeLogGroups",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
            {
              "Action": [
                "xray:PutTraceSegments",
                "xray:PutTelemetryRecords",
                "xray:GetSamplingRules",
                "xray:GetSamplingTargets",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
            {
              "Action": [
                "logs:CreateLogStream",
                "logs:PutLogEvents",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::GetAtt": [
                  "spokeSpokeAccountStateMachineSpokeAccountStateMachineLogGroupBF1216DD",
                  "Arn",
                ],
              },
            },
            {
              "Action": [
                "logs:FilterLogEvents",
                "logs:GetLogEvents",
                "logs:GetLogGroupFields",
                "logs:DescribeLogGroups",
                "logs:DescribeLogStreams",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::GetAtt": [
                  "spokeSpokeAccountStateMachineSpokeAccountStateMachineLogGroupBF1216DD",
                  "Arn",
                ],
              },
            },
            {
              "Action": [
                "logs:CreateLogDelivery",
                "logs:CreateLogStream",
                "logs:GetLogDelivery",
                "logs:UpdateLogDelivery",
                "logs:DeleteLogDelivery",
                "logs:ListLogDeliveries",
                "logs:PutLogEvents",
                "logs:PutResourcePolicy",
                "logs:DescribeResourcePolicies",
                "logs:DescribeLogGroups",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
            {
              "Action": [
                "s3:PutObject",
                "s3:PutObjectAcl",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:aws:s3:::",
                      {
                        "Ref": "BucketNameParameter",
                      },
                    ],
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:aws:s3:::",
                      {
                        "Ref": "BucketNameParameter",
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Resource": {
                "Ref": "CentralRoleArnParameter",
              },
            },
            {
              "Action": "lambda:InvokeFunction",
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "spokeMergeResultsfn262D356D",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "spokeMergeResultsfn262D356D",
                          "Arn",
                        ],
                      },
                      ":*",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": "lambda:InvokeFunction",
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "spokeSearchfn7ED063A8",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "spokeSearchfn7ED063A8",
                          "Arn",
                        ],
                      },
                      ":*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "spokeSpokeAccountStateMachineRoleDefaultPolicy4B03EE37",
        "Roles": [
          {
            "Ref": "spokeSpokeAccountStateMachineRoleEF6757B7",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "spokeSpokeAccountStateMachineRoleEF6757B7": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": {
                  "Fn::FindInMap": [
                    "ServiceprincipalMap",
                    {
                      "Ref": "AWS::Region",
                    },
                    "states",
                  ],
                },
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "Tags": [
          {
            "Key": "Solution",
            "Value": "aws-organizations-tag-inventory",
          },
          {
            "Key": "Url",
            "Value": "https://github.com/aws-samples/aws-organizations-tag-inventory",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "spokeSpokeAccountStateMachineSpokeAccountStateMachineLogGroupBF1216DD": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "LogGroupName": "/aws/statemachine/SpokeAccountStateMachine",
        "RetentionInDays": 30,
        "Tags": [
          {
            "Key": "Solution",
            "Value": "aws-organizations-tag-inventory",
          },
          {
            "Key": "Url",
            "Value": "https://github.com/aws-samples/aws-organizations-tag-inventory",
          },
        ],
      },
      "Type": "AWS::Logs::LogGroup",
      "UpdateReplacePolicy": "Delete",
    },
    "spokelayerslayerBCD68A79": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "CompatibleArchitectures": [
          "arm64",
          "x86_64",
        ],
        "Content": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "da436c1c20c60720e3c33cf202cb9d211767bef0e9f30dfb09715c1638ab637c.zip",
        },
      },
      "Type": "AWS::Lambda::LayerVersion",
      "UpdateReplacePolicy": "Delete",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
